src/filecache.c

```diff
@@ -1100,15 +1144,21 @@ static void put_return_etag(const char *path, int fd, char *etag, GError **gerr)
 
         timed_curl_easy_perform(session, &res, &response_code, &elapsed_time);
 
-        fclose(fp);
```

Comment:
It seems like it was already possible to get a good fp, then due to error skip over and end up not closing it. But by removing the fclose here in the non-error case, is there a path to not closing it. If for instance we break on non_retriable_error a couple of lines below?

Reply:
Yeah, need to re-examine this. Already not a fan of the `if (fp)` in `ifinish` -- if I jump to a region of code explicitly, that region shouldn't then need to do more checks.